common/data_refinery_common/migrations/0045_svd_algorithm.py

```diff
@@ -0,0 +1,28 @@
+from django.db import migrations, models
+
+def svd_algorithm_fix(apps, schema_editor):
+    """ Fetch existing compendia and create compendium results"""
+    CompendiumResult = apps.get_model('data_refinery_common', 'CompendiumResult')
+    ComputedFile = apps.get_model('data_refinery_common', 'ComputedFile')
+
+    compendium_computed_files = ComputedFile.objects.filter(is_compendia=True,
+                                                            result__compendium_result__isnull=False)\
```

Comment:
I think this query will be fine, but now that I've had a bit of time to think about it I don't think you need this null check necessarily. The reason I think this is that you're filtering on `is_compendia=True`, so you're already only looking at computed files that should have one and only one chain of ComputedFile->ComputationalResult->CompendiaResult, so just doing a prefetch_related should pull all of that in a single query.

However I don't think these extra filter will really hurt anything, so you can just leave it as it is if you want.

Reply:
Ok, I think we can keep it because the query looks like it's performing the Inner Join with it there but won't if it's not there.